lib/charms/mongodb/v0/shards_interface.py

```diff
@@ -0,0 +1,228 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""In this class, we manage relations between config-servers and shards.
+
+This class handles the sharing of secrets between sharded components, adding shards, and removing
+shards.
+"""
+import logging
+
+from charms.mongodb.v0.helpers import KEY_FILE
+from charms.mongodb.v0.mongodb import MongoDBConnection, NotReadyError, PyMongoError
+from charms.mongodb.v0.users import MongoDBUser, OperatorUser
+from ops.charm import CharmBase
+from ops.framework import Object
+from ops.model import BlockedStatus, MaintenanceStatus, WaitingStatus
+from tenacity import RetryError, Retrying, stop_after_delay, wait_fixed
+
+from config import Config
+
+logger = logging.getLogger(__name__)
+
+
+# The unique Charmhub library identifier, never change it
+LIBID = "55fee8fa73364fb0a2dc16a954b2fd4a"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 1
+KEYFILE_KEY = "key-file"
+OPERATOR_PASSWORD_KEY = "operator-password"
```

Comment:
is it possible to use this `MongoDBUser.get_password_key_name_for_user(username)` instead of  OPERATOR_PASSWORD_KEY  or maybe define constant like
OPERATOR_PASSWORD_KEY = MongoDBUser.get_password_key_name_for_user(username)

Reply:
good suggestion!